kyo-core/shared/src/main/scala/kyo/IO.scala

```diff
@@ -42,7 +42,7 @@ object IO:
       *   The suspended computation wrapped in an IO effect.
       */
     inline def apply[A, S](inline f: Safepoint ?=> A < S)(using inline frame: Frame): A < (IO & S) =
-        Effect.defer(f)
+        Effect.deferInline(f)
```

Comment:
This method used to inline the suspension as well when `IO` was still an `ArrowEffect`